src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -231,43 +252,72 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
-                    $this->fixPrevName($tokens, $index, $uses, $namespaceName);
-                } elseif ($tokens[$index]->isGivenKind([T_INSTANCEOF, T_NEW, CT::T_USE_TRAIT])) {
-                    $this->fixNextName($tokens, $index, $uses, $namespaceName);
-                } elseif ($tokens[$index]->isGivenKind(T_VARIABLE)) {
-                    $prevIndex = $tokens->getPrevMeaningfulToken($index);
-                    if (null !== $prevIndex && $tokens[$prevIndex]->isGivenKind(T_STRING)) {
+            $indexDiff = 0;
+            foreach (true === $this->configuration['import_symbols'] ? [true, false] : [false] as $discoverSymbolsPhase) {
+                $this->discoveredSymbols = $discoverSymbolsPhase ? [] : null;
+
+                for ($index = $namespace->getScopeStartIndex(); $index < $namespace->getScopeEndIndex() + $indexDiff; ++$index) {
+                    $origSize = \count($tokens);
+                    if ($discoverSymbolsPhase && $tokens[$index]->isGivenKind([T_CLASS, T_INTERFACE, T_TRAIT])) {
+                        $this->fixNextName($tokens, $index, $uses, $namespaceName);
+                    } elseif ($tokens[$index]->isGivenKind(T_FUNCTION)) {
+                        $this->fixFunction($functionsAnalyzer, $tokens, $index, $uses, $namespaceName);
+                    } elseif ($tokens[$index]->isGivenKind([T_EXTENDS, T_IMPLEMENTS])) {
+                        $this->fixExtendsImplements($tokens, $index, $uses, $namespaceName);
+                    } elseif ($tokens[$index]->isGivenKind(T_CATCH)) {
+                        $this->fixCatch($tokens, $index, $uses, $namespaceName);
+                    } elseif ($tokens[$index]->isGivenKind(T_DOUBLE_COLON)) {
                         $this->fixPrevName($tokens, $index, $uses, $namespaceName);
+                    } elseif ($tokens[$index]->isGivenKind([T_INSTANCEOF, T_NEW, CT::T_USE_TRAIT])) {
+                        $this->fixNextName($tokens, $index, $uses, $namespaceName);
+                    } elseif ($tokens[$index]->isGivenKind(T_VARIABLE)) {
+                        $prevIndex = $tokens->getPrevMeaningfulToken($index);
+                        if (null !== $prevIndex && $tokens[$prevIndex]->isGivenKind(T_STRING)) {
+                            $this->fixPrevName($tokens, $index, $uses, $namespaceName);
+                        }
+                    } elseif (\defined('T_ATTRIBUTE') && $tokens[$index]->isGivenKind(T_ATTRIBUTE)) { // @TODO: drop const check when PHP 8.0+ is required
+                        $this->fixNextName($tokens, $index, $uses, $namespaceName);
+                    } elseif ($discoverSymbolsPhase && !\defined('T_ATTRIBUTE') && $tokens[$index]->isComment() && Preg::match('/#\[\s*('.self::REGEX_CLASS.')/', $tokens[$index]->getContent(), $matches)) { // @TODO: drop when PHP 8.0+ is required
+                        $this->determineShortType($matches[1], $uses, $namespaceName);
```

Comment:
We need it, otherwise we might "mask" relative symbols. Consider ˙#[A]˙ attribute in `Ns` namespace and `\Another\A` class reference, such class CANNOT be shortened by adding `use Another\A` import, as it would mask it. In short, it is needed and fixes real world problems :)

Reply:
But for PHP <8.0 attribute is only a comment, so it is actually non-existing name conflict. I am not sure if we should cover scenario when users add pseudo-attributes for future-compatibility without actually importing proper symbols by themselves to ensure uniqueness. It's kind of stuff that can cause us problems IMHO 😅.

Anyway, does it have test case?

Reply:
Sure, it does have a test case and we must be defensive here, we cannot corrupt code just because user runs the fixer on PHP 7.4.